pkg/query-service/app/http_handler.go

```diff
@@ -5051,105 +5052,130 @@ func (aH *APIHandler) getQueueOverview(w http.ResponseWriter, r *http.Request) {
+	// Execute the query using the v5 querier
+	result, err := aH.Signoz.Querier.QueryRange(r.Context(), orgID, queryRangeRequest)
 	if err != nil {
+		zap.L().Error("Query execution failed", zap.Error(err))
 		apiErrObj := &model.ApiError{Typ: model.ErrorBadData, Err: err}
-		RespondError(w, apiErrObj, errQuriesByName)
+		RespondError(w, apiErrObj, nil)
 		return
 	}
 
+	result = thirdPartyApi.MergeSemconvColumns(result)
+
+	// Filter IP addresses if ShowIp is false
+	var finalResult *qbtypes.QueryRangeResponse = result
 	if !thirdPartyQueryRequest.ShowIp {
-		result = thirdPartyApi.FilterResponse(result)
+		filteredResults := thirdPartyApi.FilterResponse([]*qbtypes.QueryRangeResponse{result})
+		if len(filteredResults) > 0 {
+			finalResult = filteredResults[0]
+		}
 	}
 
-	resp := v3.QueryRangeResponse{
-		Result: result,
-	}
-	aH.Respond(w, resp)
+	// Send the response
+	aH.Respond(w, finalResult)
 }
 
+// getDomainInfo handles requests for domain information using v5 query builder
 func (aH *APIHandler) getDomainInfo(w http.ResponseWriter, r *http.Request) {
+	// Extract claims from context for organization ID
 	claims, err := authtypes.ClaimsFromContext(r.Context())
 	if err != nil {
 		render.Error(w, err)
 		return
 	}
+
 	orgID, err := valuer.NewUUID(claims.OrgID)
 	if err != nil {
 		render.Error(w, err)
 		return
 	}
 
+	// Parse the request body to get third-party query parameters
 	thirdPartyQueryRequest, apiErr := ParseRequestBody(r)
 	if apiErr != nil {
-		zap.L().Error(apiErr.Err.Error())
+		zap.L().Error("Failed to parse request body", zap.Error(apiErr.Err))
 		RespondError(w, apiErr, nil)
 		return
 	}
 
-	queryRangeParams, err := thirdPartyApi.BuildDomainInfo(thirdPartyQueryRequest)
-
-	if err := validateQueryRangeParamsV3(queryRangeParams); err != nil {
-		zap.L().Error(err.Error())
-		RespondError(w, apiErr, nil)
+	// Build the v5 query range request for domain info
+	queryRangeRequest, err := thirdPartyApi.BuildDomainInfo(thirdPartyQueryRequest)
+	if err != nil {
+		zap.L().Error("Failed to build domain info query", zap.Error(err))
+		apiErrObj := &model.ApiError{Typ: model.ErrorBadData, Err: err}
+		RespondError(w, apiErrObj, nil)
 		return
 	}
 
-	var result []*v3.Result
-	var errQuriesByName map[string]error
+	// Validate the v5 query range request
+	if err := queryRangeRequest.Validate(); err != nil {
+		zap.L().Error("Query validation failed", zap.Error(err))
+		apiErrObj := &model.ApiError{Typ: model.ErrorBadData, Err: err}
+		RespondError(w, apiErrObj, nil)
+		return
+	}
 
-	result, errQuriesByName, err = aH.querierV2.QueryRange(r.Context(), orgID, queryRangeParams)
+	// Execute the query using the v5 querier
+	result, err := aH.Signoz.Querier.QueryRange(r.Context(), orgID, queryRangeRequest)
 	if err != nil {
+		zap.L().Error("Query execution failed", zap.Error(err))
 		apiErrObj := &model.ApiError{Typ: model.ErrorBadData, Err: err}
-		RespondError(w, apiErrObj, errQuriesByName)
+		RespondError(w, apiErrObj, nil)
```

Comment:
since you are updating the API itself, update the errors to the new error structure.